charts/turing/templates/_helpers.tpl

```diff
@@ -0,0 +1,282 @@
+{{- end -}}
+
+{{- define "turing.environment" -}}
+{{- .Values.environment | default "dev" -}}
+{{- end -}}
+
+{{- define "turing.encryption.key" -}}
+{{- .Values.config.TuringEncryptionKey | default (randAlpha 12) -}}
+{{- end -}}
+
+{{- define "turing.serviceAccountName" -}}
+{{- if .Values.serviceAccount.create }}
+{{- default (include "turing.fullname" .) .Values.serviceAccount.name }}
+{{- else }}
+{{- default "default" .Values.serviceAccount.name }}
+{{- end }}
+{{- end }}
+
+{{- define "turing.mlp.encryption.key" -}}
+{{- .Values.config.MLPConfig.MLPEncryptionKey -}}
+{{- end -}}
+
+{{- define "turing.sentry.dsn" -}}
+{{- .Values.sentry.dsn -}}
+{{- end -}}
+
+{{- define "turing.plugins.directory" -}}
+/app/plugins
+{{- end -}}
+
+{{- define "turing.plugins.initContainers" -}}
+{{ if .Values.experimentEngines }}
+initContainers:
+{{ range $expEngine := .Values.experimentEngines }}
+{{ if eq (toString $expEngine.type) "rpc-plugin" }}
+- name: {{ $expEngine.name }}-plugin
+  image: {{ $expEngine.rpcPlugin.image }}
+  env:
+  - name: PLUGIN_NAME
+    value: "{{ $expEngine.name }}"
+  - name: PLUGINS_DIR
+    value: {{ include "turing.plugins.directory" . }}
+  volumeMounts:
+  - name: plugins-volume
+    mountPath: {{ include "turing.plugins.directory" . }}
+{{ end }}
+{{ end }}
+{{ end }}
+{{- end -}}
+
+{{- define "turing.initContainers" -}}
+initContainers:
+{{ with (include "turing.plugins.initContainers" . | fromYaml) -}}
+{{ if .initContainers }}
+{{- toYaml .initContainers -}}
+{{ end }}
+{{- end }}
+{{ with .Values.deployment.extraInitContainers }}
+{{- toYaml . -}}
+{{- end }}
+{{- end -}}
+
+{{- define "turing.environments" -}}
+{{ .Values.environmentConfigs | toYaml }}
+{{- end -}}
+
+{{- define "turing.environments.directory" -}}
+/app/cluster-env
+{{- end -}}
+
+{{- define "turing.environments.absolutePath" -}}
+{{- $base := include "turing.environments.directory" . -}}
+{{- $path := ternary .Values.mlp.environmentConfigSecret.envKey .Values.clusterConfig.environmentConfigPath (ne "" .Values.mlp.environmentConfigSecret.name) -}}
+{{- printf "%s/%s" $base $path -}}
+{{- end -}}
+
+{{- define "turing.get-workload-host" }}
+{{- $global := index . 0}}
+{{- $relNs := index . 1}}
+{{- $key := index . 2}}
+{{- $values := get $global $key}}
+{{- if not (hasKey $global $key) }}
+  {{- printf "" }}
+{{- else }}
+  {{- $values := get $global $key}}
+  {{- $host := "" }}
+  {{- with $values }}
+    {{- if .useServiceFqdn }}
+      {{- $host = printf "http://%s.%s.svc.cluster.local:%s%s" .serviceName $relNs .externalPort .apiPrefix}}
+    {{- else }}
+      {{- $inClusterPrefix := printf "%s%s" .vsPrefix .apiPrefix }}
+      {{- $host = printf "%s://%s%s" $global.protocol (include "common.get-external-hostname" $global) $inClusterPrefix }}
+    {{- end }}
+    {{- end }}
+  {{- printf "%s" $host }}
+  {{- end }}
+{{- end }}
+
+{{/*
+Postgres related
+*/}}
+
+{{- define "turing-postgresql.host" -}}
```

Comment:
maybe using common templates could save some lines of code here?
https://github.com/caraml-dev/helm-charts/blob/main/charts/common/templates/_names.tpl